admin/data/sql/migrations/089.do.add-checkForms-to-check.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE mtc_admin.[check] ADD checkForm_ids NVARCHAR(max) NULL;
```

Comment:
if it is for seen checkforms...
can we not ascertain that dynamically by querying the same table?  
seems a bit clunky to be storing the ids in an nvarchar column 🤔

Reply:
AIUI - it's not that simple.   This edge case is specifically for when a pupil has logged in and we detect that they have already logged in, but when we have *not* recorded that they have started the check.

The issue is that just in case they have started the check, and we failed to receive notification, so we replace the checkForm on each additional log-in attempt.

Hmm. Thinking about it, we would have to check if it plays nicely with restarts, as the expected way of checking to see if a pupil has seen a checkForm would using SQL `SELECT checkForm_id FROM check WHERE pupil_id = 1`

The alternative approach would be to create a new check for that pupil, with the new checkForm, and not flag it as a restart.

Reply:
yes, it could have further ramifications

Reply:
I've pushed changes for everything except the restart thing - that seems to cause an issue for restarts, indeed, so I have a question:

The restart service will check if a pupil can restart by comparing `checkCount` with `restartsCount`, where `checkCount` (currently) is the number of checks *started* by the pupil. I believe that we can modify the `checkCount` to check the number of *completed* checks to not be affected by this change (also, as the comment at the beginning of the `canRestart` function notes), but I’m not sure if the restart functionality will be affected. Do you think this change might cause additional problems?